plugins/discourse-local-dates/assets/javascripts/initializers/discourse-local-dates.js.es6

```diff
@@ -127,6 +132,17 @@ function buildHtmlPreview(element, siteSettings) {
   return previewsNode.outerHTML;
 }
 
+function _calculateDuration(element) {
+  const startDataset = element.parentElement.children[0].dataset;
```

Comment:
I wonder about errors if the DOM is not formated with the correct data attributes. Will it break rendering on the page if the attributes are missing or some format that can't be parsed by moment? Do we need an exception handler?

Reply:
Good point, I tried to crash it today. I realized that it can be even more bulletproof if we filter children to pick ones with date attribute. In that case, <div> or any other element between will not affect.

```javascript
function _rangeElements(element) {
  return Array.from(element.parentElement.children).filter(
    (span) => span.dataset.date
  );
}
```

.dataset is returning [DomStringMap](https://developer.mozilla.org/en-US/docs/Web/API/DOMStringMap) object which is forgivable `domStringMap.fdsfdsfdsfds // => undefined` so previous version was not crashing but didn't display popup correctly.

Each individual date from date range is parsed individually and returns invalid date if something goes wrong.

My most comprehensive test example is:
```
- [date=2021-33-44 timezone="Australia/Sydney"] → <div>test</div>[date=2021-12-14 time=14:30 timezone="Australia/Sydney"]<div>test</div>
```

In that case, only second date is used for popup:
<img width="250" alt="Screen Shot 2021-09-17 at 10 28 47 am" src="https://user-images.githubusercontent.com/72780/133705611-ff09d9af-c303-49f6-8d37-32d46bd0fead.png">